woocommerce/woocommerce-ios
# [Backend Receipts] Rename existing receipts while replacement system is introduced

Closes: #11826 
#11837 needs to be merged first.

## Description
This PR renames the current locally-generated receipt objects and references to `Legacy...`, since we're introducing new backend-generated receipts that eventually will replace the current implementation. 

* Legacy receipts: Only supported for card present payment methods, saved locally only (not shared across devices or simulators).
* Backend receipts: Supports all payment methods, available through all devices.

Both systems (local and backend) must be retained in the codebase for the time being since one or the other might be used based on merchant WooCommerce versions, with this rename me attempt to avoid conflicts and make easier to navigate the code while building the new system on top and deal with the receipts logic.

Since locally-generated receipts will be removed entirely from the codebase after a couple of months, this change also intends to make its removal cleaner/easier later on.

We left methods from `ReceiptAction` outside of this renaming since the method signature will be enough to differentiate, but we might change it later if needed.

## Testing instructions
We introduce no changes to the logic except renaming, the only risk is the view controller's rename since we [load the view from a nib based on the class name](https://github.com/woocommerce/woocommerce-ios/blob/18b5df706b42f4a6544306ea2ef78d6ed0be27a6/WooCommerce/Classes/ViewRelated/CardPresentPayments/LegacyReceiptViewController.swift#L19), and names must match to avoid a runtime crash.

* On a store that accepts WooPayments, go through the card present payment flow (pay an order by card, either with a reader or by enabling`-simulate-stripe-card-reader`) and save the receipt when prompted.
* Refresh the order by navigating back and re-opening the order, or triggering a pull to refresh.
* Observe the "See receipt" button, tap on it and confirm that the receipt is rendered and can be viewed as expected.

## Screenshots

![2024-01-30 existing see receipt flow](https://github.com/woocommerce/woocommerce-ios/assets/3812076/ea8951e5-4b69-4476-a815-698498a2151c)
